src/main/java/com/minecolonies/coremod/commands/CommandEntryPointNew.java

```diff
@@ -0,0 +1,596 @@
+    }
+
+    private Map<String, TreeNode<Menu>> getNavigationCommands(@NotNull final TreeNode<Menu> parentTreeNode)
+    {
+        final Map<String, TreeNode<Menu>> map = new HashMap<>();
+        final List<TreeNode<Menu>> children = parentTreeNode.getChildren();
+        for (final TreeNode<Menu> treeNode : children)
+        {
+            map.put(treeNode.getData().getMenuItemName().toLowerCase(), treeNode);
+        }
+        return map;
+    }
+
+    private Map<String, ActionMenuHolder> getActionCommands(@NotNull final TreeNode<Menu> treeNode, @NotNull final List<ActionMenuHolder> parsedHolders)
+    {
+        final Map<String, ActionMenuHolder> map = new HashMap<>();
+        final Menu menu = treeNode.getData();
+        if (!menu.getMenuType().isNavigationMenu())
+        {
+            final ActionMenu actionMenu = (ActionMenu) menu;
+            for (final ActionArgument actionArgument : actionMenu.getActionArgumentList())
+            {
+                if (usedThisActionArgumentName(parsedHolders, actionArgument))
+                {
+                    continue;
+                }
+                map.put(actionArgument.getName() + ":", new ActionMenuHolder(treeNode, actionArgument));
+            }
+        }
+        return map;
+    }
+
+    private boolean usedThisActionArgumentName(@NotNull final List<ActionMenuHolder> parsedHolders, @NotNull final ActionArgument actionArgument)
+    {
+        for (final ActionMenuHolder actionMenuHolder : parsedHolders)
+        {
+            if (actionMenuHolder.getActionArgument().getName().equals(actionArgument.getName()))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @NotNull
+    @Override
+    public List<String> getTabCompletions(
+                                           @NotNull final MinecraftServer server,
+                                           @NotNull final ICommandSender sender,
+                                           @NotNull final String[] args,
+                                           @Nullable final BlockPos pos)
+    {
+        final ParsingResult parsingResult = getTabCompletionsAndParsingHolders(root, server, sender, args, pos);
+        return parsingResult.getTabCompletions();
+    }
+
+    @NotNull
+    private ParsingResult getTabCompletionsAndParsingHolders(
+                                           @NotNull final TreeNode<Menu> treeNode,
+                                           @NotNull final MinecraftServer server,
+                                           @NotNull final ICommandSender sender,
+                                           @NotNull final String[] args,
+                                           @Nullable final BlockPos pos)
+    {
+        if (treeNode.getData().getMenuType().isNavigationMenu())
+        {
+            final Map<String, TreeNode<Menu>> childs = getNavigationCommands(treeNode);
+            final String lowerCaseArg0 = args[0].toLowerCase();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define the locale to be used in this String operation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1449)

Reply:
this